### ISSUE TITLE
ERROR: VALIDATION on examples

### ISSUE THREAD

**Issue by nixpulvis** [ash-rs/ash#917]
After getting setup with the VulkanSDK on macOS, I'm getting a lot of these errors at runtime when running either of the examples.

```
ERROR:
VALIDATION [VUID-vkAcquireNextImageKHR-semaphore-01779 (1461184347)] : Validation Error: [ VUID-vkAcquireNextImageKHR-semaphore-01779 ] Object 0: handle = 0xd5b26f0000000010, type = VK_OBJECT_TYPE_SEMAPHORE; | MessageID = 0x5717e75b | vkAcquireNextImageKHR():  Semaphore must not have any pending operations. The Vulkan spec states: If semaphore is not VK_NULL_HANDLE it must not have any uncompleted signal or wait operations pending (https://vulkan.lunarg.com/doc/view/1.3.280.0/mac/1.3-extensions/vkspec.html#VUID-vkAcquireNextImageKHR-semaphore-01779)
```

I haven't tested this on Linux so I can't say if it's my environment or something in ash itself, but I thought I'd open this issue since I don't see what else I should change without changing source code.

**Comment by MarijnS95** [comment]
@nixpulvis Yup I've seen this as well (on Linux), the example is violating some semaphore rules that are caught in newer validation layers. It's not a problem with `ash` itself (since it doesn't implement any Vulkan logic) but it is a problem with the example that should be fixed.

**Comment by nixpulvis** [comment]
@MarijnS95 good to know, thanks.

I'm *just* now starting out learning Vulkan (for the second time). So if I don't give up again and figure it out I'll be sure to come back around with a PR. No promises tho :P

**Comment by filnet** [comment]
Could also be a VVL false positive. Timeline semaphores validation is currently broken.

See https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues?q=is%3Aissue+is%3Aopen+semaphore

**Comment by MarijnS95** [comment]
@filnet swapchain functionality doesn't support the usage of timeline semaphores.

**Comment by BennicX** [comment]
I was getting the exact same error on my program but I was able to make it go away by creating the semaphore inside the same closure as the swapchain_loader.acquire_next_image() function.
```
        let _ = base.render_loop(|| {
            let semaphore = base.device.create_semaphore(&vk::SemaphoreCreateInfo::default(), None)
            .expect("error creating the semaphore");
            let (present_index, _) = base
                .swapchain_loader
                .acquire_next_image(
                    base.swapchain,
                    std::u64::MAX,
                    semaphore,
                    vk::Fence::null(),
                )
                .unwrap();
```

Not sure if this is a hack or an actual solution to this problem.

**Comment by vadim-s-kovy** [comment]
There is a simple fix for that: https://github.com/ash-rs/ash/pull/979
